airflow/providers/hashicorp/common/vault_client.py

```diff
@@ -0,0 +1,392 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import List, Optional
+
+import hvac
+from cached_property import cached_property
+from hvac.api.auth_methods import github, kubernetes, ldap, radius, userpass
+from hvac.api.secrets_engines import aws, azure, gcp, kv_v1, kv_v2
+from hvac.exceptions import InvalidPath, VaultError
+from requests import Response
+
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+VALID_KV_VERSIONS: List[int] = [1, 2]
+VALID_AUTH_TYPES: List[str] = [
+    'approle',
+    'aws_iam',
+    'azure',
+    'github',
+    'gcp',
+    'kubernetes',
+    'ldap',
+    'radius',
+    'token',
+    'userpass'
+]
+
+
+class VaultClient(LoggingMixin):  # pylint: disable=too-many-instance-attributes
+    """
+    Retrieves Authenticated client from Hashicorp Vault.
+
+    :param url: Base URL for the Vault instance being addressed.
+    :type url: str
+    :param auth_type: Authentication Type for Vault. Default is ``token``. Available values are in
+        :py:const:`airflow.providers.hashicorp.common.vault_client.VALID_AUTH_TYPES`.
+    :type auth_type: str
+    :param mount_point: The "path" the secret engine was mounted on. Default depends on the engine used.
+          For "aws_iam" the default mount_point is "aws", for "token" it is "secret" and for
+          all other authentication methods it's the same as the authentication method.
```

Comment:
Let's separate one for Secrets Engine (https://www.vaultproject.io/docs/secrets) and one for Authentication (https://www.vaultproject.io/docs/auth)

Reply:
I think I now understa YEp.